### ISSUE TITLE
Unclear if block size for -S option is 2^n bits or 2^n bytes

### ISSUE THREAD

**Issue by sgmihai** [mhx/dwarfs#86]
The mkdwarfs help says that:
  -S [ --block-size-bits ] arg          block size bits (size = 2^bits)
But I tried to run something like :  mkdwarfs -i v64 -o n64S30B8.dwarf -N10 -L4g -S30 -B8
and it tells me : low memory limit (4 GiB), need 10 GiB to efficiently compress 1 GiB blocks with 10 threads
But 2^30 bits in mib is 128 MiB, not 1GiB, so 8 times less.
Which one is right then, 2^n bits or bytes block size ?
Also, the block size info only appears when there is not enough memory for it, it would be nice to have it display before starting comrpression, with the others stats.

**Comment by mhx** [comment]
Ohhh, I see where the confusion comes from. :)

This should probably say:

> -S [ --block-size-bits ] arg block size bits (size = 2^arg bytes)

Would that make things more clear?

I didn't consider it an issue because if the option was referring to the *literal* number of bits in a block, you wouldn't specify it as an exponent. Also, it'd probably say "block bits" and not "block size bits".

> Also, the block size info only appears when there is not enough memory for it, it would be nice to have it display before starting comrpression, with the others stats.

I'll see if I can squeeze it in somewhere.

**Comment by sgmihai** [comment]
Right, that makes sense, should've been obvious in hindsight, but I guess it could use clearing up, as well as saying what the default value is (= ), as with other options.
Also:
for -S31 it says "need 16 EiB to efficiently compress 16 EiB blocks"
for -S32 it doesn't say anything (probably overflows ?)

**Comment by mhx** [comment]
Nice catch, yep, these are both overflowing. I'll fix this and probably put an upper limit on the block size as well to avoid overflows with 64 bit ints. :)
